### PR TITLE
feat: add service monitor

### PR DESCRIPTION
**Pull request by tao12345666333**
Signed-off-by: Jintao Zhang <zhangjintao9020@gmail.com>

fixes: #76

### CODE REVIEW

File: charts/apisix-ingress-controller/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) .Values.serviceMonitor.enabled }}
```
Comment (tokers) [inline charts/apisix-ingress-controller/templates/servicemonitor.yaml:1]: Should add the Apache License Header.

File: charts/apisix-ingress-controller/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "apisix-ingress-controller.fullname" . }}
+  {{- if .Values.serviceMonitor.namespace }}
+  namespace: {{ .Values.serviceMonitor.namespace }}
+  {{- end }}
+spec:
+  endpoints:
+  - targetPort: http
```
Comment (tokers) [inline charts/apisix-ingress-controller/templates/servicemonitor.yaml:11]: Note in newer APISIX releases, APISIX using a separate port to expose Prometheus. So the `targetPort` might not be `http`, also, the Service definition should also be adjusted.

**Review by tokers — APPROVED** [review]

File: charts/apisix-ingress-controller/templates/servicemonitor.yaml (hunk 1 above)
Comment (tao12345666333) [reply charts/apisix-ingress-controller/templates/servicemonitor.yaml:1]: OK

File: charts/apisix-ingress-controller/templates/servicemonitor.yaml (hunk 2 above)
Comment (tao12345666333) [reply charts/apisix-ingress-controller/templates/servicemonitor.yaml:27]: This port is only used to expose the metrics of apisix-ingress-controller itself, not APISIX

Comment (tokers) [reply charts/apisix-ingress-controller/templates/servicemonitor.yaml:27]: Got it.

**Review by gxthrj — APPROVED** [review]

**Pull request merged**